src/lua/api-gateway/validation/validator.lua

```diff
@@ -122,58 +128,51 @@ end
+    local ok, redisread = redisConnectionProvider:getConnection(redis_RO_upstream)
     if ok then
         local redis_key, selecterror = redisread:hget(key, hash_field)
         redisread:set_keepalive(30000, 100)
         if (type(redis_key) == 'string') then
             return redis_key
         end
     else
-        ngx.log(ngx.WARN, "Failed to read key " .. tostring(key) .. " from Redis cache:[", redis_host, ":", redis_port, "]. Error:", err)
+        ngx.log(ngx.WARN, "Failed to read key " .. tostring(key))
     end
     return nil;
 end
 
 
 -- is wrapper over redis exists  but returns boolean instead
 function BaseValidator:exists(key)
-    local redisread = redis:new()
-    local redis_host, redis_port = self:getRedisUpstream()
-    local ok, err = redisread:connect(redis_host, redis_port)
-    if err then
-        ngx.log(ngx.WARN, "Failed to connect to redis for ".. key .." host:", redis_host, ".Error:", err)
-        return false
-    end
-
-    local redis_key, selecterror = redisread:exists(key)
-    redisread:set_keepalive(30000, 100)
-    if selecterror or redis_key ~= 1 then
-        ngx.log(ngx.WARN, "Failed to read key ".. key .." from Redis cache:", redis_host, ".Error:", err)
+    local ok, redisread = redisConnectionProvider:getConnection();
+    if ok then
+        local redis_key, selecterror = redisread:exists(key)
+        redisread:set_keepalive(30000, 100)
+        if selecterror or redis_key ~= 1 then
+            ngx.log(ngx.WARN, "Failed to read key " .. key .. " from Redis cache:", redis_host, ".Error:", err)
+            return false
+        end
+        return true;
+    else
+        ngx.log(ngx.WARN, "Failed to perform exists on key " .. tostring(key))
         return false
     end
-
-    return true;
 end
 
 -- saves a value into the redis cache. --
 -- the method uses HSET redis command --
 -- it retuns true if the information is saved in the cache, false otherwise --
 function BaseValidator:setKeyInRedis(key, hash_name, keyexpires, value)
-    ngx.log(ngx.DEBUG, "Storing in Redis the key [", tostring(key), "], expireat=", tostring(keyexpires), ", value=", tostring(value) )
-    local rediss = redis:new()
-    local redis_host, redis_port = self:getRedisUpstream(self.redis_RW_upstream)
-    local ok, err = rediss:connect(redis_host, redis_port)
+    ngx.log(ngx.DEBUG, "Storing in Redis the key [", tostring(key), "], expireat=", tostring(keyexpires), ", value=", tostring(value))
+    local ok, rediss = redisConnectionProvider:getConnection(redis_RW_upstream)
     if ok then
         --ngx.log(ngx.DEBUG, "WRITING IN REDIS JSON OBJ key=" .. key .. "=" .. value .. ",expiring in:" .. (keyexpires - (os.time() * 1000)) )
         rediss:init_pipeline()
         rediss:hset(key, hash_name, value)
         if keyexpires ~= nil then
             rediss:pexpireat(key, keyexpires)
         end
-        local commit_res, commit_err = rediss:commit_pipeline()
+        local _, commit_err = rediss:commit_pipeline()
```

Comment:
It's an unused return value and the code was filled with them, which I only find cluttering.